### DISCUSSION TITLE
Does patchInitialised() always get called multiple times?

### DISCUSSION THREAD

**Discussion by cristianvogel** [cables-gl/cables_docs#693] [Answers & Questions] [answered]
In my application, which is supposed to be embeddind just one Cables instance , when I run dev , I've noticed that `onPatchLoaded` gets call multiple times ( 10 to be precise ) . I am working with Svelte 4.0 on the app side.

```svelte
  function patchInitialized(patch: any) {
    // implement callback function used by Cables op.callback
    patch.config.interactionFromUI = function (params: string[]) {
      $UI_StateArray[$CurrentPickedID].toggle();
      $UI_StateArray = $UI_StateArray; // reactive assignment
    };
    // get all the variables from the Cables patch
    $CablesParams = patch.getVars();
    ($ConsoleText = "Patch initialized"), Object.keys($CablesParams);
    console.count("Patch initialized");
  }
```

Possible suspect is that call to `patch.getVars()`

**Comment by cristianvogel** [comment]
I tried moving my patchLoaded functionality to `onFinishedLoading()`  and then it only happens once. 

```svelte
   onPatchLoaded: () => {},
   onFinishedLoading: () => {
          patchInitialized(get(CablesPatch));
          patchFinishedLoading();
        }
        
        ....
        
        
```
If I put `console.count` as the onPatchLoaded callback function , I get 10 counts

**Comment by steam0r** [comment, marked as answer]
this event is called whenever cables "deserializes" content, i guess you are using subpatchops. then this event is called once per finished subpatchop.
